backend/src/main/java/com/bakdata/conquery/sql/conversion/query/TableExportQueryConverter.java

```diff
@@ -101,106 +106,99 @@ private static QueryStep convertPrerequisite(TableExportQuery exportQuery, Conve
+		if (validityDate == null) {
 			return Optional.of(ColumnDateRange.empty());
 		}
-		SqlFunctionProvider functionProvider = context.getSqlDialect().getFunctionProvider();
-		ColumnDateRange validityDate = functionProvider.forValidityDate(table.findValidityDate());
+		final SqlFunctionProvider functionProvider = context.getSqlDialect().getFunctionProvider();
+		final ColumnDateRange validityDateRange = functionProvider.forValidityDate(validityDate);
 		// when exporting tables, we want the validity date as a single-column daterange string expression straightaway
-		Field<String> asStringExpression = functionProvider.encloseInCurlyBraces(functionProvider.daterangeStringExpression(validityDate));
+		final Field<String> asStringExpression = functionProvider.encloseInCurlyBraces(functionProvider.daterangeStringExpression(validityDateRange));
 		return Optional.of(ColumnDateRange.of(asStringExpression).asValidityDateRange(alias));
 	}
 
 	private static List<FieldWrapper<?>> initializeFields(CQTable cqTable, Map<Column, Integer> positions) {
 
-		Field<?>[] exportColumns = createPlaceholders(positions, cqTable);
+		final Field<?>[] exportColumns = createPlaceholders(positions, cqTable);
 		for (Column column : cqTable.getConnector().getTable().getColumns()) {
 			// e.g. date column(s) are handled separately and not part of positions
 			if (!positions.containsKey(column)) {
 				continue;
 			}
-			int position = positions.get(column) - 1;
+			final int position = positions.get(column) - POSITION_OFFSET;
 			exportColumns[position] = createColumnSelect(column, position);
 		}
 
 		return Arrays.stream(exportColumns).map(FieldWrapper::new).collect(Collectors.toList());
 	}
 
+	private static Table<Record> joinConnectorTableWithPrerequisite(
+			CQTable cqTable,
+			SqlIdColumns ids,
+			QueryStep convertedPrerequisite,
+			CDateRange dateRestriction,
+			ConversionContext context
+	) {
+		final SqlFunctionProvider functionProvider = context.getSqlDialect().getFunctionProvider();
+		final Table<Record> connectorTable = DSL.table(DSL.name(cqTable.getConnector().getTable().getName()));
+		final Table<Record> convertedPrerequisiteTable = DSL.table(DSL.name(convertedPrerequisite.getCteName()));
+
+		final ColumnDateRange validityDate = functionProvider.forValidityDate(cqTable.findValidityDate());
+		final List<Condition> joinConditions =
+				Stream.concat(
+						ids.join(convertedPrerequisite.getQualifiedSelects().getIds()).stream(),
+						Stream.of(functionProvider.dateRestriction(functionProvider.forCDateRange(dateRestriction), validityDate))
+				).toList();
+
+		return functionProvider.innerJoin(connectorTable, convertedPrerequisiteTable, joinConditions);
+	}
+
 	private static Field<?>[] createPlaceholders(Map<Column, Integer> positions, CQTable cqTable) {
 
-		Field<?>[] exportColumns = new Field[positions.size() + 1];
+		final Field<?>[] exportColumns = new Field[TableExportQuery.calculateWidth(positions) - POSITION_OFFSET];
```

Comment:
@awildturtok Hier hatte nur die -1 gefehlt. Ich habe das nochmal in eine Konstante mit Erklärung gepackt, damit dass nicht irgendwann untergeht.